### Display Sample Data

In [23]:
%run '../lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


In [24]:
dfFlt = load_df('dfFlt')
cats = get_var_cats()

#### Display Sample Data

In [25]:
dfFlt.select(*cats['orgVars']).distinct().toPandas()

org_id
0  349115
1  360421
2  360541

- 3 organizations

In [26]:
dfFlt.filter( F.col('org_id') == 360421).select(*cats['sectionVars']).distinct().toPandas()

section_id
0       498961
1       499023
2       537231
3       498945
4       498977
5       498971
6       498965
7       499094
8       499047
9       530503
10      499028
11      527034
12      498941
13      498967
14      522071
15      509045
16      499081
17      520208

- 18 sections in organization 360421

In [27]:
dfFlt.filter( F.col('section_id') == 499028).select(*cats['learnerVars']).distinct().toPandas()

learner_id
0      1259558
1      1248382
2      1248826
3      1249701
4      1248402
5      1248370
6      1257687
7      1248404
8      1248792
9      1248400
10     1248786
11     1249343
12     1248378
13     1248403
14     1248371
15     1248369
16     1249244
17     1248377
18     1251888
19     1255484
20     1248372
21     1248788
22     1249507
23     1259171
24     1249243
25     1258474

- 25 learners in section 499028

In [28]:
dfFlt.filter( F.col('section_id') == 499028).select(*cats['assessmentVars']).distinct().toPandas()

assessment_id  assessment_instance_id
0         748408                  799468
1         747201                  926453
2         748403                  799457
3         750551                  820432
4         747722                  799442
5         748402                  907587
6         751391                  842344
7         748402                  907549
8         750014                  923506
9         750043                  799474

- 9 assessments in section 499028
- one instance per assessment per section

###### Display Assignments for Section 498941

In [29]:
dfFlt.filter( F.col('section_id') == 499028).select('assessment_id', *cats['assignmentVars']).distinct().orderBy('assignment_due_date').toPandas()

assessment_id assignment_due_date assignment_final_submission_date  \
0         750043 2019-10-31 04:59:00              2019-10-31 04:59:00   
1         747722 2019-11-02 04:59:00              2019-11-02 04:59:00   
2         748402 2019-11-02 04:59:00              2019-11-02 04:59:00   
3         748403 2019-11-09 05:59:00              2019-11-09 05:59:00   
4         748408 2019-12-15 05:59:00              2019-12-15 05:59:00   
5         750551 2020-01-11 05:59:00              2020-01-11 05:59:00   
6         748402 2020-01-11 05:59:00              2020-01-11 05:59:00   
7         750014 2020-01-23 05:59:00              2020-01-23 05:59:00   
8         751391 2020-02-28 05:59:00              2020-02-28 05:59:00   
9         747201 2020-03-11 04:59:00              2020-03-11 04:59:00   

   assignment_late_submission  assignment_max_attempts assignment_start_date  \
0                       False                        1   2019-10-24 05:00:00   
1                       False                        1   2019-10-25 05:00:00   
2                       False                        1   2019-10-31 05:00:00   
3                       False                        1   2019-10-30 05:00:00   
4                       False                        1   2019-12-11 06:00:00   
5                       False                        1   2019-12-12 06:00:00   
6                       False                        1   2019-10-23 05:00:00   
7                       False                        1   2020-01-10 06:00:00   
8                       False                        1   2020-02-26 06:00:00   
9                       False                        1   2020-03-09 05:00:00   

  ced_assignment_type_code  is_affecting_grade max_student_stop_datetime  \
0               assessment                True   2019-10-28 16:24:37.517   
1               assessment                True   2019-10-25 17:07:25.717   
2               assessment                True   2019-10-31 17:29:37.994   
3               assessment                True   2019-10-31 17:23:19.626   
4               assessment                True   2019-12-13 19:49:42.369   
5               assessment                True   2020-01-09 18:03:36.016   
6               assessment                True   2020-01-09 19:36:23.312   
7               assessment                True   2020-01-21 14:52:26.890   
8               assessment                True   2020-02-28 05:59:49.336   
9               assessment                True   2020-03-11 05:00:10.856   

  min_student_start_datetime  number_of_distinct_instance_items  \
0    2019-10-24 16:36:52.673                                  6   
1    2019-10-25 16:33:59.288                                 20   
2    2019-10-31 16:45:03.226                                 10   
3    2019-10-30 17:19:55.740                                 18   
4    2019-12-11 19:28:13.214                                 19   
5    2019-12-18 19:49:03.145                                  5   
6    2019-10-29 18:36:22.562                                 10   
7    2020-01-10 17:22:13.248                                 18   
8    2020-02-26 17:15:42.037                                 39   
9    2020-03-09 18:14:20.612                                 20   

   number_of_learners  
0                  26  
1                  26  
2                   1  
3                  26  
4                  26  
5                  26  
6                  26  
7                  26  
8                  26  
9                  26

- 9 assignments (one for each assessment) in section 499028

###### Display Assignment Items for Assessment 748408 of Section 499028

In [30]:
dfFlt.filter( F.col('assessment_id') == 748408).select(*cats['itemVars']).distinct().toPandas()

item_is_offline_scored item_type_code_name scoring_type_code
0                   False               essay            manual
1                   False         shortAnswer            manual
2                   False      multipleChoice         automatic
3                   False            matching         automatic
4                   False               essay      [unassigned]
5                   False           trueFalse         automatic

- 5 items in assessment 721235
- 1 item (true/false) is automatic scored
- 2 item essay is one manual scored the other [unassigned]
- 1 item short answer is manual scored
- 1 item matching is automatic scored

###### Display Assignment Attempts for Assessment 748408

In [31]:
dfFlt.filter( F.col('assessment_id') == 748408 ).select(*cats['assignmentAttemptVars']).distinct().toPandas()

assessment_instance_attempt_id  assignment_attempt_number  \
0                          1231041                          1   
1                          1148290                          1   
2                          1110794                          1   
3                          1227978                          1   
4                          1155442                          1   
5                          1098979                          1   
6                          1122674                          1   
7                          1175820                          1   
8                          1117904                          1   
9                          1190008                          1   
10                         1165346                          1   
11                         1098986                          1   
12                         1125030                          1   
13                         1239925                          1   
14                         1098981                          1   
15                         1193414                          1   
16                         1210783                          1   

    final_score_unweighted  is_force_scored  is_manual_scoring_required  \
0                       38            False                       False   
1                       44            False                       False   
2                       23            False                       False   
3                       32            False                       False   
4                       46            False                       False   
5                       39            False                       False   
6                       40            False                       False   
7                       44            False                       False   
8                       43            False                       False   
9                       44            False                       False   
10                      20            False                       False   
11                      41            False                       False   
12                      39            False                       False   
13                      45            False                       False   
14                      37            False                       False   
15                      11            False                       False   
16                      29            False                       False   

    learner_assignment_attempt_id learner_attempt_status  \
0                         1002687           fully scored   
1                         1002584           fully scored   
2                         1002575           fully scored   
3                         1002702           fully scored   
4                         1002648           fully scored   
5                         1002598           fully scored   
6                         1002614           fully scored   
7                         1002831           fully scored   
8                         1002637           fully scored   
9                         1002558           fully scored   
10                        1002669           fully scored   
11                        1002763           fully scored   
12                        1002798           fully scored   
13                        1002807           fully scored   
14                        1002587           fully scored   
15                        1002681           fully scored   
16                        1002814           fully scored   

    points_possible_unweighted     scored_datetime  student_start_datetime  \
0                         48.0 2019-12-13 10:21:57 2019-12-13 17:11:12.994   
1                         48.0 2019-12-13 10:17:50 2019-12-13 17:11:19.226   
2                         48.0 2020-01-08 07:08:16 2019-12-13 17:11:40.970   
3                         48.0 2020-01-09 09:59:44 2019-12-13 17:11:20.474   
4          

- 16 attempts
- Majority of students started (student_start_datetime) and stopped (student_stop_datetime) on 2019-12-13
- Scored (was_fully_scored_datetime) between 12/12 and 1/10

###### Display Assignment Attempt Items for Attempt 1098979

In [33]:
dfFlt.filter( F.col('assessment_instance_attempt_id') == 1098979 ).select(*cats['itemAttemptVars']).distinct().toPandas()

assessment_item_response_id assigned_item_status  is_deleted  \
0                       1548819               scored       False   
1                       1645778               scored       False   
2                       1614874               scored       False   
3                       1648977               scored       False   
4                       1589534               scored       False   
5                       1657844               scored       False   
6                       1572928               scored       False   
7                       1554352               scored       False   
8                       1622899               scored       False   
9                       1570752               scored       False   
10                      1650527               scored       False   
11                      1619280               scored       False   
12                      1599166               scored       False   
13                      1626774               scored       False   
14                      1635215               scored       False   
15                      1569829               scored       False   
16                      1627583               scored       False   
17                      1577219               scored       False   
18                      1600362               scored       False   

    learner_assigned_item_attempt_id response_correctness  
0                            1526587              correct  
1                            1526589              correct  
2                            1526598              correct  
3                            1438642              correct  
4                            1526590              correct  
5                            1526586              correct  
6                            1526592              correct  
7                            1526585              correct  
8                            1526596    partially_correct  
9                            1526593              correct  
10                           1526581              correct  
11                           1526582            incorrect  
12                           1526584              correct  
13                           1526597              correct  
14                           1526591            incorrect  
15                           1526594            incorrect  
16                           1526583              correct  
17                           1526595              correct  
18                           1526588              correct

- 18 items
- All answered (scored)
  - 14 correct
  - 3 incorrect
  - on partially correct


###### Display Assignment Attempt Items for Attempt 1193414

In [34]:
dfFlt.filter( F.col('assessment_instance_attempt_id') == 1193414 ).select(*cats['itemAttemptVars']).distinct().toPandas()

assessment_item_response_id assigned_item_status  is_deleted  \
0                       1662814               scored       False   
1                       1556132               scored       False   
2                       1642668               scored       False   
3                       1575456               scored       False   
4                       1624295               scored       False   
5                       1563437               scored       False   
6                       1552133               scored       False   
7                       1609927               scored       False   
8                       1631174               scored       False   
9                       1583454               scored       False   
10                      1555799               scored       False   
11                      1654912               scored       False   
12                      1559832               scored       False   
13                      1635745               scored       False   
14                      1610358               scored       False   
15                      1623177               scored       False   
16                      1607672               scored       False   
17                      1617860               scored       False   
18                      1567338               scored       False   

    learner_assigned_item_attempt_id response_correctness  
0                            1400105              correct  
1                            1400108            incorrect  
2                            1400094            incorrect  
3                            1400110            incorrect  
4                            1400101            incorrect  
5                            1400114            incorrect  
6                            1400096              correct  
7                            1400099              correct  
8                            1400095            incorrect  
9                            1400104            incorrect  
10                           1400097            incorrect  
11                           1400100            incorrect  
12                           1400111              correct  
13                           1400109              correct  
14                           1400112              correct  
15                           1400103            incorrect  
16                           1400098            incorrect  
17                           1400113              correct  
18                           1400102            incorrect

- 18 items
- All answered (scored)
  - 7 correct
  - 11 incorrect
